### ISSUE TITLE
Got error while performing test after training your model for a batch size of 100. 

### ISSUE THREAD

**Issue by GauravKrRoy** [JaywongWang/DenseVideoCaptioning#19]
Here I am attaching the screenshot of the generated error and in the left you can see the checkpoints generated after training the model.
![Screenshot (10)](https://user-images.githubusercontent.com/35898224/59579999-065e4100-90ed-11e9-96fc-fb70d63df2d0.png)

**Comment by JaywongWang** [comment]
What is the value of options['init_from'] (defined in opt.py) ? Have you properly set it to the checkpoint file?

**Comment by GauravKrRoy** [comment]
I forgot to pass the checkpoint in options['init_from'] (defined in opt.py) but 
I had replaced following code of lines in test.py
print('Restoring model from %s'%options['init_from'])
    saver = tf.train.Saver()
    saver.restore(sess, options['init_from'])
with these lines of code:
print('Restoring model from %s'%options['init_from'])
    saver = tf.train.Saver()
    saver.restore(sess, './checkpoints/1/epoch00_0.98_00_lr0.001000.ckpt')
and its' working fine. Is it the correct replacement?

**Comment by JaywongWang** [comment]
It is correct.

**Comment by GauravKrRoy** [comment]
I have question regarding anchors.txt file in get_anchors.py
Here, you are finding the length of each timestamp for every video and then KMEANS clustering is applied to generate the anchors. Can u please clear it.

**Comment by JaywongWang** [comment]
The anchor mechanism is firstly introduced by Faster-RCNN. The designed anchors help discretize the prediction space. In object detection the anchors refer to boxes with different ratios (e.g., 1:1, 1:2, 2:1, 2D space), while in temporal proposals the anchors represent different temporal spans or lengths (1-D). KMEANS clustering is to obtain the K most frequent anchors. At each time step t, the model predicts the corresponding confidences for the K pre-defined anchors.

**Comment by GauravKrRoy** [comment]
How anchor based approach for proposal generation can be generalised for caption generation of new longer video as same anchors.txt file is being used for training as well as testing.

**Comment by JaywongWang** [comment]
Hard to be generalized. This is one of the main weaknesses for anchor-based approaches. It can be partly solved by anchor-free model.

**Comment by GauravKrRoy** [comment]
Is there any possibility to apply anchor free approach in your code. If yes, then please explain.

**Comment by GauravKrRoy** [comment]
Is there any possibility to apply anchor free approach in your code. If yes, then please explain.

**Comment by GauravKrRoy** [comment]
Is there any possibility to apply anchor free approach in your code. If yes, then please explain.

**Comment by JaywongWang** [comment]
Our model sorely relies on the anchor-based approach (SST). It does not support anchor-free approaches. You may try a two-step scheme: first generate proposals using anchor-free approach (e.g., BSN [1]), then apply simple captioning method (e.g., TA [2]) to obtain the corresponding caption.

[1] Tianwei Lin et al. Bsn: Boundary sensitive network for temporal action proposal generation. In ECCV, 2018.
[2] Li Yao et al. Describing videos by exploiting temporal structure. In ICCV, 2015.

**Comment by GauravKrRoy** [comment]
Thanks a lot.

**Comment by aurooj** [comment]
> What is the value of options['init_from'] (defined in opt.py) ? Have you properly set it to the checkpoint file?

@JaywongWang  Hay, thanks for sharing your work with us. I was trying to generate captions for my own videos, and tried to run your test script after doing the following steps:
1. Extracted C3D feats
2. Applied PCA from activitynet to get 500-dimensional features.
3. Saved features in hdf5 file.
4. In test.py script, I set the paths to my hdf5 file and used my own test_ids json file.
However, the code crashes when I am trying to load your pre-trained model.ckpt file. 
I changed the following in opt.py file:
options['init_from'] = 'checkpoints/model.ckpt' 
options['train_proposal'] = False   # whether to train variables of proposal module
    options['train_caption'] = False    # whether to train variables of captioning module
    options['evaluate_metric'] = False  # whether refer to evalutaion metric (CIDEr, METEOR, ...) for optimization
(because i only want to generate captions without any training for my videos)
But the test.py crashes at line 79 with following error:
Restoring model from checkpoints/model.ckpt
W tensorflow/core/framework/op_kernel.cc:993] Not found: Tensor name "caption_module/multi_rnn_cell/cell_0/lstm_cell/weights" not found in checkpoint files checkpoints/model.ckpt
	 [[Node: save/RestoreV2_11 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save/Const_0, save/RestoreV2_11/tensor_names, save/RestoreV2_11/shape_and_slices)]]
W tensorflow/core/framework/op_kernel.cc:993] Not found: Tensor name "caption_module/multi_rnn_cell/cell_0/lstm_cell/weights" not found in checkpoint files checkpoints/model.ckpt

I am new to tensorflow, but the only thing which I can get out of it is that the keys for different layers don't match in the model definition and pretrained model.
Any help will be highly appreciated.

**Comment by momalave** [comment]
Did you figure out the issue? Were you able to run the model on your own videos?

**Comment by KPKWCE** [comment]
> Here I am attaching the screenshot of the generated error and in the left you can see the checkpoints generated after training the model.
> ![Screenshot (10)](https://user-images.githubusercontent.com/35898224/59579999-065e4100-90ed-11e9-96fc-fb70d63df2d0.png)

Hello GauravKrRoy and JaywongWang, Here I am attaching the error I got while training train.py, Please help for this
![1](https://user-images.githubusercontent.com/26180175/92987652-3b209a00-f4e2-11ea-88b5-86d0a5c4a1f3.PNG)


Thank you in advance...
